x/auth/types/account.go

```diff
@@ -222,6 +222,10 @@ func (ma ModuleAccount) Validate() error {
 		return fmt.Errorf("address %s cannot be derived from the module name '%s'", ma.Address, ma.Name)
 	}
 
+	if ma.BaseAccount == nil {
```

Comment:
Actually, this check should be before ma.Address, as a ModuleAccount inherits a BaseAccount.
The tests still panic as of now.

Reply:
Interesting, it didn't panic on mine locally! Thanks @julienrbrt, updated/fixed.